src/algebra/jordan/basic.lean

```diff
@@ -0,0 +1,223 @@
+/-
+Copyright (c) 2021 Christopher Hoskin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christopher Hoskin
+-/
+import algebra.ring.basic
+import algebra.lie.of_associative
+import data.real.basic
+import linear_algebra.basic
+
+/-!
+# Jordan algebras
+
+Let `A` be a non-associative algebra (i.e. a module equipped with a bilinear multiplication
+operation). Then `A` is said to be a (commutative, linear) Jordan algebra if the multiplication is
+commutative and satisfies a weak associativity law known as the Jordan Identity: for all `a` and `b`
+in `A`,
+```
+(a * b) * a^2 = a * (b * a^2)
+```
+i.e. the operators of multiplication by `a` and `a^2` commute. Every associative algebra can be
+equipped with a second  multiplication making it into a commutative Jordan algebra.
+Jordan algebras arising this way are said to be special. There are also exceptional Jordan algebras
+which can be shown not to be the symmetrization of any associative algebra. The 3x3
+matrices of octonions is the canonical example.
+
+Every Jordan algebra `A` has a triple product defined, for `a` `b` and `c` in `A` by
+```
+⦃a b c⦄ = (a * b) * c - (a * c) * b + a * (b * c).
+```
```

Comment:
I think it might be worth adding a `jordan_triple` definition to capture this.

Reply:
Ha! My motivation for creating this PR is that I want to define Jordan (*-)triples in Lean and show that they are power associative, which requires first proving the same result for Jordan algebras. In fact I already have a branch on my computer with some of the theory developed, but I thought it best to work on one PR at a time (besides, I will need to rework some of what I have already done based on what I have learnt through the development of this PR).

Reply:
I've made a start here: https://github.com/leanprover-community/mathlib/tree/jordan-triples (needs more work).

Reply:
Now a PR https://github.com/leanprover-community/mathlib/pull/11553